packages/core/docs/advanced/caching.md

```diff
@@ -1,99 +1,133 @@
+
 ```sh
 yarn add @vue-storefront/cache
+yarn add <DRIVER-NAME> # eg. @vsf-enterprise/redis-cache
 ```
 
-## How it works?
+### Configuration
 
-## Invalidators
+The next step is to register `@vue-storefront/cache` package as a module in `nuxt.config.js` with driver and invalidation configuration.
 
-```ts
-const commercetoolsProductInvalidator = (response) => {
-  return response.products.map(p => p.id)
-}
+::: warning
+Make sure this package is added to the `modules` array, not `buildModules`.
+:::
 
-const commercetoolsCategoryInvalidator = (response) => {
-  return response.categories.map(p => c.id)
-}
+```javascript
+// nuxt.config.js
+
+export default {
+  modules: [
+    ['@vue-storefront/cache/nuxt', {
+      invalidation: {
+        endpoint: '/cache-invalidate',
+        key: 'CHANGE_THIS'
+      },
+      driver: [
+        '<DRIVER NAME>',
+        {
+          driverConfig
+        }
+      ]
+    }]
+  ]
+};
 ```
 
-## The driver
+We can break down package configuration into two pieces:
 
+* `invalidation` (optional object) - contains URL to invalidate cache and key used to prevent unauthorized access.
+* `driver` (array or string) - contains the path to or name of the driver package and its configuration. If the driver doesn't require any configuration, you can pass a string instead of an array. Please refer to the documentation of the driver you are using for more information.
 
-```ts
-const CacheDriver = (options, invalidators) => {
-  const exampleCache = new Redis(options);
+###  Tags
 
-  return {
-    invoke: async ({ getTags, render }) => {
-      if (exampleCache.hasCache()) {
-        return exampleCache.getCache()
-      }
+If you follow the above steps and run the application, you won't see any performance difference. This is because only pages with tags are cached.
 
-      const content = await render();
-      exampleCache.storeCache(getTags(), content)
-      return content;
-    },
-    invalidate: ({ req, res }) => {
-      invalidators.forEach(invalidator => invalidator(req))
-    }
-  };
-};
-```
+Please refer to the [Using tags](#using-tags) section for more information.
 
-## Core mechanism
+## How it works?
 
+When the page is requested, the cache driver checks if there is an already rendered page in the cache matching the current route. If there is, it will serve the cached version. Otherwise:
 
-```ts
-import { useCache } from '@vue-storefront/cache';
-import redisDriver from '@vue-storefront/cache-redis'
+1. Current page will be rendered on the server.
+2. If the page contains tags, the result will be saved in the cache.
+3. Page is served to the user.
 
-const cache = useCache(); // Inside setup
-cache.addTags([{ prefix: 'P', value: '123' }]) // adds tags
-cache.setTags((currentTags) => [...currentTags, { prefix: 'P', value: '123' }]) // adds tags, also you can skip previous one
-cache.getTags(); // returns current tags
-```
+## Using tags
```

Comment:
How about splitting this into:
"Tagging content" - theory + how to tag content
"Invalidating cached content"